packages/ckeditor5-engine/src/conversion/downcastdispatcher.js

```diff
@@ -166,9 +165,12 @@ export default class DowncastDispatcher {
 	 * @fires attribute
 	 * @param {module:engine/model/range~Range} range The inserted range.
 	 * @param {module:engine/view/downcastwriter~DowncastWriter} writer The view writer that should be used to modify the view document.
+	 * @param {Object} [options={}] Additional, custom configuration passed to the converters through
+	 * {module:engine/conversion/downcastdispatcher~DowncastConversionApi conversion api}. Used only with `dataDowncast` group.
 	 */
-	convertInsert( range, writer ) {
+	convertInsert( range, writer, options = {} ) {
```

Comment:
I am not sure I like this parameter here -- because you have to then document it the way you did. Maybe we could handle it somehow differently?

For example, set a property of `conversionApi` at the beginning of `.get()` and clear at the end? Hmmm.

Or maybe it is fine. But I'd change the documentation. In this place of API docs noone is interested where this parameter is used. This is kind of integration detail. We can write in the tutorial that options passed in `.get()` are later available in `conversionApi` but what you described is too much.

Reply:
Correct me if I'm wrong, but in the current form, we'd need to have an additional param in every event and method.

In that case, conversionApi would be better as it's already widely available.

Reply:
>  we'd need to have an additional param in every event and method
Not in every, but in `convertInsert()` and `convertMarkerAdd()`. To those methods that are used in `DataController#toView()`.

Reply:
For me, it sounds good to set options directly to conversionApi (at the beginning of `DataController#toView()`) and remove when the conversion is done (at the end of `DataController#toView()`)